informer/informer.go

```diff
@@ -95,6 +96,8 @@ func New(config Config) (*Informer, error) {
 		resyncPeriod: config.ResyncPeriod,
 	}
 
+	prometheus.MustRegister(prometheus.Collector(newInformer))
```

Comment:
I would like to do this in the framework. We have the informer there. So when the framework boots we should register the metric. I do not like to do these magic things in the constructors a lot.

Reply:
I tried this, but we get `informer.Interface` passed into the framework. I can't cast that interface to `prometheus.Collector`. Am I missing something?

Reply:
For me it makes more sense to register it here in the constructor. Why does the `controller` package have to even know about it?